### PR TITLE
limited OSRM native routing API

### PR DESCRIPTION
**Pull request by XavDmz**
# Limited OSRM native routing API for road2

Linked to issue #87.
This PR covers only routing request and responses with OSRM formalism, and using an OSRM source engine. 

OSRM query parameters, mandatory or not, are used if they already had an equivalent in road2's code.
Other **optional** parameters are ignored.
Aside from these ignored paramters, the request and response are compatible with OSRM specifications.

`routeRequest` has been updated with a property named `engineExtras`, a dictionary designed to contain engine response properties used by an API other than "simple". As an example, OSRM waypoint's distance to requested coordinates is such a property.
`osrmSource` has been updated to set `routeRequest.engineExtras` with this kind of additional properties, necesssary to craft OSRM native API response.

### CODE REVIEW

**Comment by XavDmz** [comment]
The current state (https://github.com/IGNF/road2/pull/98/commits/d9d931b9cd442b4ddeea7370cff3ed5b97942016) has only been testing on a singular routing request, with intermediates coordinates / waypoints.
The getcapabilties functionality, and both user and developper documentation, have not yet been updated.
The former needs a rework to generate the GetCapabilities content dynamically, which will allow for easier porting to other APIs, as well as live update of capabilties description.

**Review by azarz — APPROVED** [review]
Looks great to me!

File: src/js/sources/osrmSource.js [hunk 1]
```diff
@@ -372,6 +373,11 @@ module.exports = class osrmSource extends Source {
     let profile;
     let optimization;
     let routes = new Array();
+    let engineExtras = {
```
Comment (lgrd) [inline src/js/sources/osrmSource.js:376]: See if this code is the final HTTP response status

File: src/js/responses/routeResponse.js [hunk 2]
```diff
@@ -189,4 +192,27 @@ module.exports = class routeResponse extends Response {
     this._routes = st;
   }
 
+    /**
+  *
+  * @function
+  * @name get engineExtras
+  * @description Récupérer les propriétés non génériques, spécifiques au moteur et à son API
```
Comment (lgrd) [inline src/js/responses/routeResponse.js:199]: Add a little doc

**Review by lgrd — APPROVED** [review]
Thanks ! ;)

**Pull request merged**